scalafix-tests/src/test/resources/checkSyntax/NoAutoTupling.source

```diff
@@ -0,0 +1,89 @@
+rewrites = [NoAutoTupling]
+<<< add explicit tuple
+object tup {
+  def fooo(t: (Int, Int)): Int = ???
+  fooo(1, 2)
+}
+>>>
+object tup {
+  def fooo(t: (Int, Int)): Int = ???
+  fooo((1, 2))
+}
+
+<<< multiple parameter lists, all auto-tupled
+object tup {
+  def fooo(t: (Int, Int))(s: (String, String)): Int = ???
+  fooo(1, 2)("a", "b")
+
+  def baar(t: (Boolean, Int, String))(s: (Boolean, String))(k: (Int, String)): String = ???
+  baar(true, 1, "foo")(false, "42")(42, "foo")
+}
+>>>
+object tup {
+  def fooo(t: (Int, Int))(s: (String, String)): Int = ???
+  fooo((1, 2))(("a", "b"))
+
+  def baar(t: (Boolean, Int, String))(s: (Boolean, String))(k: (Int, String)): String = ???
+  baar((true, 1, "foo"))((false, "42"))((42, "foo"))
+}
+
+<<< multiple parameter lists, some auto-tupled
+object tup {
+  def fooo(t: (Int, Int))(s: (String, String)): Int = ???
+  fooo((1, 2))("a", "b")
+
+  def baar(t: (Boolean, Int, String))(s: (Boolean, String))(k: (Int, String)): String = ???
+  baar(true, 1, "foo")((false, "42"))(42, "foo")
+}
+>>>
+object tup {
+  def fooo(t: (Int, Int))(s: (String, String)): Int = ???
+  fooo((1, 2))(("a", "b"))
+
+  def baar(t: (Boolean, Int, String))(s: (Boolean, String))(k: (Int, String)): String = ???
+  baar((true, 1, "foo"))((false, "42"))((42, "foo"))
+}
+
+<<< already tupled calls stay the same
+object tup {
+  def fooo(t: (Int, Int)): Int = ???
+  fooo((1, 2))
+}
+>>>
+object tup {
+  def fooo(t: (Int, Int)): Int = ???
+  fooo((1, 2))
+}
+
+<<< methods not involving tuples stay the same
+object tup {
+  def sum(a: Int, b: Int): Int = ???
+  sum(1, 2)
+}
+>>>
+object tup {
+  def sum(a: Int, b: Int): Int = ???
+  sum(1, 2)
+}
+
+<<< methods with tuples, but not single parameter
+object tup {
+  def sum(a: Int, b: (Int, String)): Int = ???
+  sum(1, (2, "foo"))
+}
+>>>
+object tup {
+  def sum(a: Int, b: (Int, String)): Int = ???
+  sum(1, (2, "foo"))
+}
+
+<<< auto-tupling with lambdas
```

Comment:
I OK with adding a SKIP to this test for now. It might be good to add another similar test that does currying

```scala
@ def f: (((Int, String)) => ((String, List[Int])) => Int) = a => b => a._1
defined function f
@ f(1, "")("a", Nil) // original
res17: Int = 1
@ f((1, ""))(("a", Nil)) // expected
res16: Int = 1
```
because the current implementation uses `.zip` which will not work across curried anonymous functions.

However, as long as the rewrite does nothing in these cases and documents this limitation then it's OK for now. I opened an issue in scalameta https://github.com/scalameta/scalameta/issues/837 but I don't see a reason to block this PR until that issue is fixed. This rewrite is already very useful as long as it does not have false positives.